components/homme/src/share/interpolate_mod.F90

```diff
@@ -1177,9 +1181,12 @@ subroutine setup_latlon_interp(elem,interpdata,par)
              sphere2_xyz = spherical_to_cart( ref2sphere(cart%x,cart%y,     &
                   elem(ii)%corners3D,cubed_sphere_map,elem(ii)%corners,elem(ii)%facenum ))
              sphere_xyz = spherical_to_cart(sphere)
+!$OMP CRITICAL
              err=max(err,distance(sphere2_xyz,sphere_xyz))
```

Comment:
Could use a reduction instead of critical region. In any case, I recommend you compute `distance` outside of the critical region, then do `err=max(err,d))` inside, thus minimizing the computations in the critical region.